src/gui/folderstatusmodel.cpp

```diff
@@ -413,8 +413,8 @@ FolderStatusModel::SubFolderInfo *FolderStatusModel::infoForIndex(const QModelIn
 FolderStatusModel::SubFolderInfo *FolderStatusModel::infoForFileId(const QByteArray& fileId, SubFolderInfo* info) const
 {
   const QVector<SubFolderInfo>& infoVec = info ? info->_subs : _folders;
-  for(int i = 0, end = infoVec.size(); i < end; i++) {
-    auto *info = const_cast<SubFolderInfo *>(&infoVec[i]);
+  for(const auto &i : infoVec) {
+    auto *info = const_cast<SubFolderInfo *>(&i);
```

Comment:
This function totally creeps me out for several reasons... Moreover, it is never used. Any objections to simply remove this method (for clean history in a separate commit before the modernizing loop commit)?

Reply:
Yeah, don't like that method either... And indeed unused, good catch. I'll kill it.